### PR TITLE
Add first-class subview support

### PR DESCRIPTION
**Pull request by scotttrinh**
This change adds "views into the entity store" as a first-class concept beyond the `knownIds` list.

This should allow for fetching subsets of a collection (pagination, filtered views, etc).

### `view`

For collections that opt into this feature, `knownIds` stops being a useful view into the data unless you want to just see what IDs have been fetched from somewhere, so definitely use `viewAt` instead of `view` to get your populated views.

### Removal

`removeAt` does _not_ remove the entity from the store or from knownIds, so if you want to fully remove a resource from the store, you can do:

```ts
existingCollection.removeAt('parentId', 'someId').remove('someId');
```

That will remove it from both the ID map at the given key, and the underlying entity store.

### CODE REVIEW

File: src/__tests__/at.ts [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+    'Returns the successful items'
+  );
+});
+
+test('#viewAt, with no items loaded', t => {
+  const col = new Collection<Item>();
+  t.deepEqual(col.viewAt('parentId'), RD.initial, 'Returns initial');
+});
+
+test('#viewAt, with a some successes and a failure', t => {
```
Comment (skswanke) [inline src/__tests__/at.ts:209]: Great coverage!

**Review by skswanke — APPROVED** [review]
The changes here look really good! The unit test coverage is great as well. One functional question.
> removeAt does not remove the entity from the store or from knownIds
Is there a method to remove the entity directly, rather than going through a parent id?

**Review by alekhinen — APPROVED** [review]
sweet, this lgtm!

**Comment by scotttrinh** [comment]
@skswanke 

> Is there a method to remove the entity directly, rather than going through a parent id?

If you want to remove a resource from the store, it's a bit of a two step process:

1. Remove it from whatever ID lists it might be a part of.
2. Remove it from the entity store itself.

I feel like we need this flexibility to support using a key to show a filtered view where we want to remove something from one of those views, but not remove it from the store altogether.

Perhaps there can be a shortcut method that just does both? Maybe `removeAll`? It would need to iterate through each `idMap` and look for the ID.

**Comment by skswanke** [comment]
@scotttrinh Ok, yeah I was hoping for the world where the consumer doesn't need to keep track of all of the places a resource might be, but I suppose it would be a pretty significant performance hit to loop through everything.

**Pull request merged**